benchmarking/commons/benchmark_definitions/yahpo.py

```diff
@@ -199,21 +226,133 @@ def yahpo_iaml_xgboost_benchmark(dataset_name, metric, mode="max"):
+yahpo_iaml_benchmark = {
+    "rpart": yahpo_iaml_rpart_benchmark,
+    "glmnet": yahpo_iaml_glmnet_benchmark,
+    "ranger": yahpo_iaml_ranger_benchmark,
+    "xgboost": yahpo_iaml_xgboost_benchmark,
 }
 
 
-def yahpo_rbv2_rpart_benchmark(dataset_name, metric, mode="max"):
+yahpo_iaml_instances = ["40981", "41146", "1489", "1067"]
+
+
+yahpo_iaml_selected_instances = {
+    "rpart": {
+        "f1": ["41146", "1489", "1067"],
+        "auc": ["41146", "1489", "1067"],
+    },
+    "glmnet": {
+        "f1": ["1489", "1067"],
+        "auc": ["1489"],
+    },
+    "ranger": {
+        "f1": ["41146", "1489", "1067"],
+        "auc": ["41146", "1489"],
+    },
+    "xgboost": {
+        "f1": ["40981", "1489"],
+        "auc": ["40981", "1489"],
+    },
+}
+
+
+def yahpo_iaml_benchmark_definitions(method, restrict_fidelities=False) -> dict:
+    return {
+        benchmark_name("iaml", method, metric, name): yahpo_iaml_benchmark[method](
+            name,
+            metric,
+            mode,
+            restrict_fidelities=restrict_fidelities,
+        )
+        for metric, mode in yahpo_iaml_metrics
+        for name in yahpo_iaml_instances
+    }
+
+
+def yahpo_iaml_selected_benchmark_definitions(
+    method, restrict_fidelities=False
+) -> dict:
+    return {
+        benchmark_name("iaml", method, metric, name): yahpo_iaml_benchmark[method](
+            name,
+            metric,
+            mode,
+            restrict_fidelities=restrict_fidelities,
+        )
+        for metric, mode in yahpo_iaml_metrics
+        for name in yahpo_iaml_selected_instances[method][metric]
+    }
+
+
+def yahpo_iaml_rpart_benchmark_definitions(restrict_fidelities=False) -> dict:
+    return yahpo_iaml_benchmark_definitions("rpart", restrict_fidelities)
+
+
+def yahpo_iaml_rpart_selected_benchmark_definitions(restrict_fidelities=False) -> dict:
+    return yahpo_iaml_selected_benchmark_definitions("rpart", restrict_fidelities)
+
+
+def yahpo_iaml_glmnet_benchmark_definitions(restrict_fidelities=False) -> dict:
+    return yahpo_iaml_benchmark_definitions("glmnet", restrict_fidelities)
+
+
+def yahpo_iaml_glmnet_selected_benchmark_definitions(restrict_fidelities=False) -> dict:
+    return yahpo_iaml_selected_benchmark_definitions("glmnet", restrict_fidelities)
+
+
+def yahpo_iaml_ranger_benchmark_definitions(restrict_fidelities=False) -> dict:
+    return yahpo_iaml_benchmark_definitions("ranger", restrict_fidelities)
+
+
+def yahpo_iaml_ranger_selected_benchmark_definitions(restrict_fidelities=False) -> dict:
+    return yahpo_iaml_selected_benchmark_definitions("ranger", restrict_fidelities)
+
+
+def yahpo_iaml_xgboost_benchmark_definitions(restrict_fidelities=False) -> dict:
+    return yahpo_iaml_benchmark_definitions("xgboost", restrict_fidelities)
+
+
+def yahpo_iaml_xgboost_selected_benchmark_definitions(
+    restrict_fidelities=False,
+) -> dict:
+    return yahpo_iaml_selected_benchmark_definitions("xgboost", restrict_fidelities)
+
```

Comment:
Is it necessary to spell out all of those combinations? I have not seen them being used anywhere. Maybe it would more consise to `modify yahpo_iaml_benchmark_definitions` to the form of
```
def yahpo_iaml_benchmark_definitions(method, restrict_fidelities=False) -> dict:
    assert method in yahpo_iaml_benchmark, f"method must be one of {yahpo_iaml_benchmark.keys()}"
```

and do the same with `yahpo_iaml_selected_benchmark_definitions`. This way the definitions will be easier to extend.

Reply:
Makes sense